src/vs/workbench/contrib/search/browser/searchActionsTextQuickAccess.ts

```diff
@@ -29,6 +34,59 @@ registerAction2(class TextSearchQuickAccessAction extends Action2 {
 
 	override async run(accessor: ServicesAccessor, match: RenderableMatch | undefined): Promise<any> {
 		const quickInputService = accessor.get(IQuickInputService);
-		quickInputService.quickAccess.show(TEXT_SEARCH_QUICK_ACCESS_PREFIX);
+		const searchText = getSearchText(accessor);
+		const searchContents = searchText ? ' ' + searchText : '';
+		quickInputService.quickAccess.show(TEXT_SEARCH_QUICK_ACCESS_PREFIX + searchContents);
 	}
 });
+
+function getSearchText(accessor: ServicesAccessor): string | null {
+	const editorService = accessor.get(IEditorService);
+	const configurationService = accessor.get(IConfigurationService);
+
+	if (editorService.activeTextEditorControl === undefined) {
+		return null;
+	}
+	const activeEditor: IEditor = editorService.activeTextEditorControl;
+	if (!activeEditor) {
+		return null;
+	}
+	if (!activeEditor.hasTextFocus()) {
+		return null;
+	}
+
+	// only happen if it would also happen for the search view
+	const seedSearchStringFromSelection = configurationService.getValue<IEditorOptions>('editor').find!.seedSearchStringFromSelection;
+	if (!seedSearchStringFromSelection) {
+		return null;
+	}
+
+	const range = activeEditor.getSelection();
+	if (!range) {
+		return null;
+	}
+	if (!isCodeEditor(activeEditor) || !activeEditor.hasModel()) {
+		return null;
+	}
+
+	let searchText = '';
+	for (let i = range.startLineNumber; i <= range.endLineNumber; i++) {
+		let lineText = activeEditor.getModel().getLineContent(i);
+		if (i === range.endLineNumber) {
+			lineText = lineText.substring(0, range.endColumn - 1);
+		}
+
+		if (i === range.startLineNumber) {
+			lineText = lineText.substring(range.startColumn - 1);
+		}
+
+		if (i !== range.startLineNumber) {
+			lineText = '\n' + lineText;
+		}
+
+		searchText += lineText;
+	}
```

Comment:
If both searchView and this use the same way to get the current selection, I would prefer that you abstract out the behavior from the searchview into an exported function such that it can be reused here.

Reply:
Good idea, I will try to modify it

Reply:
I abstracted this method and put it in findController.ts, not sure if this is suitable